tests/e2e/tests/e2e/GitSelfSignCert.spec.ts

```diff
@@ -0,0 +1,105 @@
+ const browserTabsUtil: BrowserTabsUtil = e2eContainer.get(CLASSES.BrowserTabsUtil);
+ const editor: Editor = e2eContainer.get(CLASSES.Editor);
+
+ const workspaceName = 'gitSelfSignCert';
+
+ suite('Checking git + self sign cert', async () => {
+     const workspacePrefixUrl: string = `${TestConstants.TS_SELENIUM_BASE_URL}/dashboard/#/ide/${TestConstants.TS_SELENIUM_USERNAME}/`;
+     const gitProjectUrl = `https://${TestConstants.TS_SELF_HOSTED_GIT_SERVER_URL}/maxura/gitService.git`;
+     const committedFile = 'README.md';
+
+     suiteSetup(async function () {
+         const wsConfig = await testWorkspaceUtils.getBaseDevfile();
+         wsConfig.metadata!.name = workspaceName;
+         await browserTabsUtil.navigateTo(TestConstants.TS_SELENIUM_BASE_URL);
+         await loginPage.login();
+         await testWorkspaceUtils.createWsFromDevFile(wsConfig);
+     });
+
+     test('Wait until created workspace is started', async () => {
+         await dashboard.waitPage();
+         await browserTabsUtil.navigateTo(workspacePrefixUrl + workspaceName);
+         await ide.waitWorkspaceAndIde();
+         await projectTree.openProjectTreeContainer();
+         await driverHelper.wait(TimeoutConstants.TS_SELENIUM_LOAD_PAGE_TIMEOUT);
+     });
+
+
+     test('Clone project', async () => {
+         await cloneTestRepo(gitProjectUrl);
+
+         await projectTree.waitProjectImportedNoSubfolder('gitService');
+         await projectTree.waitItem('gitService/README.md');
+     });
+
+     test('Change commit and push', async function changeCommitAndPushFunc() {
+        const currentDate: string = Date.now().toString();
+        await projectTree.expandPathAndOpenFile('gitService', committedFile);
+        await editor.type(committedFile, currentDate + '\n', 1);
+        await gitPlugin.openGitPluginContainer();
+        await gitPlugin.waitChangedFileInChagesList(committedFile);
+        await gitPlugin.stageAllChanges(committedFile);
+        await gitPlugin.waitChangedFileInChagesList(committedFile);
+        await gitPlugin.typeCommitMessage(this.test!.title + currentDate);
+        await gitPlugin.commitFromCommandMenu();
+        await gitPlugin.pushChangesFromCommandMenu();
+        await quickOpenContainer.typeAndSelectSuggestion('git-admin', 'Git: ' + `https://${TestConstants.TS_SELF_HOSTED_GIT_SERVER_URL}` + ' (Press \'Enter\' to confirm your input or \'Escape\' to cancel)');
+        await quickOpenContainer.typeAndSelectSuggestion('admin', 'Git: https://git-admin@' + `${TestConstants.TS_SELF_HOSTED_GIT_SERVER_URL}` + ' (Press \'Enter\' to confirm your input or \'Escape\' to cancel)');
+        await gitPlugin.waitDataIsSynchronized();
+    });
+
+ });
+
+ suite('Stopping and deleting the workspace', async () => {
+    test(`Stop and remove workspace`, async () => {
+        await workspaceHandlingTests.stopAndRemoveWorkspace(workspaceName);
+    });
+});
+
+ async function cloneTestRepo(linkToRepo: string) {
```

Comment:
Would it be useful also for some other test or is it specific for just this one test case? If another test could benefit from this method, I would suggest moving it to some library. If it's specific to just this test case, I'm OK leaving it here.

Reply:
Very similar function exists in GitSsh and GitPublishBranch tests. But for GitSelfSignCert cloneTestRepo() is specific in getting repo url step.